lib/lwt/lwt_promise.ml

```diff
@@ -0,0 +1,18 @@
+open Js_of_ocaml
+open Promise
+
+let to_lwt (p : 'a promise Js.t) : 'a Lwt.t =
+  let (r, w) = Lwt.task () in
+  ignore
+    ((Js.Unsafe.coerce p)##_then
+       (Js.wrap_callback (fun v -> Lwt.wakeup w v))
+       (Js.wrap_callback (fun e -> Lwt.wakeup_exn w (try raise e with e -> e))));
```

Comment:
If this is the `onReject` callback, it looks like it is capable of rejecting Lwt promises with any value, because JS promises can be rejected with any value, but Lwt promises can only be rejected with OCaml exceptions `exn`.